<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <node pkg="image_transport" type="republish" name="republish" output="screen" args="compressed in:=/ovc/rgb/image_color out:=/asoom/img"/>

  <node pkg="asoom" type="asoom_node" name="asoom" output="screen">
    <!-- Settings for adjusting perf -->
    <param name="stereo_use_sgbm" value="False"/>
    <param name="pose_graph_num_frames_opt" value="30"/>
    <param name="map_thread_period_ms" value="5000"/>
    <param name="map_req_point_density" value="100"/>
    <param name="map_resolution" value="5"/>

    <param name="require_imgs" value="True"/>
    <param name="use_semantics" value="False"/>
    <param name="use_gps_stamp" value="False"/>
    <param name="semantic_lut_path" value="$(find asoom)/config/semantic_lut.yaml"/>
    <param name="pose_graph_fix_scale" value="False"/>
    <param name="keyframe_dist_thresh_m" value="2"/>

    <param name="rectifier_calib_path" value="$(find asoom)/config/titan_wide.yaml"/>
    <remap from="~pose" to="/orbslam3_ros_node/pose"/>
    <remap from="~gps" to="/ublox/fix"/>
    <!--param name="rectifier_calib_path" value="$(find asoom)/config/aerial_pennov.yaml"/>
    <remap from="~pose" to="/sfm_pose"/>
    <remap from="~img" to="/ovc/bonus_left/left/image_color"/>
    <remap from="~sem" to="/ovc/bonus_left/left/image_label"/-->
  </node>
</launch>
